internal/api/registry/manifests.go

```diff
@@ -198,8 +198,8 @@ func (a *API) handleGetOrHeadManifest(w http.ResponseWriter, r *http.Request) {
 		w.Write(manifestBytes)
 	}
 
-	//count the pull
-	if r.Method == http.MethodGet && r.Header.Get("X-Keppel-No-Count-Towards-Last-Pulled") != "1" {
+	//count the pull unless a special header is set or the token was issued by the janitor user identity used by the trivy proxy
+	if r.Method == http.MethodGet && r.Header.Get("X-Keppel-No-Count-Towards-Last-Pulled") != "1" && authz.UserIdentity.UserType() != keppel.JanitorUser {
```

Comment:
I tested the expanded test by removing this change again and it generated multiple test failures with similar pattern to:

```
Δ tmp/easypg-diff3125603906/expected ⟶   tmp/easypg-diff3125603906/actual

─────┐
• 1: │
─────┘
    ⋮  1 │UPDATE manifests SET last_pulled_at = 6000 WHERE repo_id = 1 AND digest = 'sha256:0962993cac41a5429d58b5142279ea849c3291cdffcc881d0188f1be73928ffd';
    ⋮  2 │UPDATE manifests SET last_pulled_at = 6000 WHERE repo_id = 1 AND digest = 'sha256:44e1f93f1d7e65ecd552c9b3544af5966095177dbc3b3757fb2e7629dc219f1a';
    ⋮  3 │UPDATE manifests SET last_pulled_at = 6000 WHERE repo_id = 1 AND digest = 'sha256:4c4f2bca300e74786a04590aa15cfcbfa1f3ec64c15fad0a0df8a6674dcbf34b';
    ⋮  4 │UPDATE manifests SET last_pulled_at = 6000 WHERE repo_id = 1 AND digest = 'sha256:e3c1e46560a7ce30e3d107791e1f60a588eda9554564a5d17aa365e53dd6ae58';
  1 ⋮  5 │UPDATE trivy_security_info SET vuln_status = 'Critical', next_check_at = 9600, checked_at = 6000, check_duration_secs = 0 WHERE repo_id = 1 AND digest = 'sha256:0962993cac41a5429d58b5142279ea849c3291cdffcc881d0188f1be73928ffd';
  2 ⋮  6 │UPDATE trivy_security_info SET vuln_status = 'Critical', next_check_at = 9600, checked_at = 6000, check_duration_secs = 0 WHERE repo_id = 1 AND digest = 'sha256:44e1f93f1d7e65ecd552c9b3544af5966095177dbc3b3757fb2e7629dc219f1a';
  3 ⋮  7 │UPDATE trivy_security_info SET vuln_status = 'Critical', next_check_at = 9600, checked_at = 6000, check_duration_secs = 0 WHERE repo_id = 1 AND digest = 'sha256:4c4f2bca300e74786a04590aa15cfcbfa1f3ec64c15fad0a0df8a6674dcbf34b';
--- FAIL: TestCheckVulnerabilitiesForNextManifest (0.58s)
    manifests_test.go:649: exit status 1
```